public/app/features/alerting/unified/api/alertingApi.ts

```diff
@@ -0,0 +1,47 @@
+import { BaseQueryFn, createApi } from '@reduxjs/toolkit/query/react';
+import { lastValueFrom } from 'rxjs';
+
+import { BackendSrvRequest, getBackendSrv } from '@grafana/runtime';
+
+import { PromBuildInfoResponse } from '../../../../types/unified-alerting-dto';
+
+import { discoverAlertmanagerFeatures } from './buildInfo';
+
+const backendSrvBaseQuery = (): BaseQueryFn<BackendSrvRequest> => async (requestOptions) => {
+  try {
+    const response = await lastValueFrom(getBackendSrv().fetch<PromBuildInfoResponse>(requestOptions));
+
+    return { data: response.data };
+  } catch (error) {
+    return {
+      error: {
+        data: error,
+      },
+    };
+  }
+};
+
+export const alertingApi = createApi({
+  reducerPath: 'alertingApi',
+  baseQuery: backendSrvBaseQuery(),
+  endpoints: (build) => ({
+    discoverAmFeatures: build.query({
```

Comment:
one last remark; perhaps we could split up each endpoint in to its own function so we don't have to inline it in the `createApi` function

Reply:
Yes, I think we can use the `injectEndpoints` method to provide endpoints separation without much toil

Reply:
Usually, we shouldn't use `queryFn` to construct the API call. This case is exceptional because the `discoverFeatures` function makes multiple API calls to discover available features.